elf/pinning.go

```diff
@@ -45,7 +45,14 @@ func pinObject(fd int, namespace, object, name string) error {
 		return fmt.Errorf("bpf fs not mounted at %q", BPFFSPath)
 	}
 	mapPath := filepath.Join(BPFFSPath, namespace, object, name)
-	err = os.MkdirAll(filepath.Dir(mapPath), 0755)
+	if err := pinObjectCommon(fd, mapPath); err != nil {
+		return err
+	}
+	return nil
+}
+
+func pinObjectCommon(fd int, mapPath string) error {
```

Comment:
Can `PinObjectGlobal` maybe use `PinObjectCustom`, i.e. is `pinObject` still needed?